benchmark/kmeans.py

```diff
@@ -169,35 +259,163 @@ def _check_same_fit(self, estimator, name, max_iter, assert_allclose):
+            "Please also refer to the sklearn documentation page at "
+            "https://scikit-learn.org/stable/modules/generated/sklearn.cluster.KMeans.html "  # noqa
+            "for more informatiom on the CLI parameters."
+        )
+    )
+
+    argparser.add_argument(
+        "--sample-weight", default="random", choices=["unary", "random", "None"]
+    )
+
+    argparser.add_argument("--init", default="random", choices=["k-means++", "random"])
+
+    argparser.add_argument("--n-clusters", default=127, type=int)
+
+    argparser.add_argument("--max-iter", default=100, type=int)
+
+    argparser.add_argument(
+        "--dataset",
+        default="spoken-arabic-digit",
+        choices=["spoken-arabic-digit"],
+        help="Name of the dataset used to benchmark the engine providers.",
+    )
+
+    argparser.add_argument(
+        "--n-data-samples",
+        default=None,
+        type=int,
+        help=(
+            "n_data_samples samples will be sampled from the dataset. If necessary, "
+            "some samples will be randomly repeated. If None, the original amount "
+            "of samples in the dataset will be used without alteration."
+        ),
+    )
+
+    argparser.add_argument(
+        "--n-features",
+        default=None,
+        type=int,
+        help=(
+            "n_features features will be sampled from the dataset. If necessary, "
+            "some features will be randomly repeated. If None, the original amount "
+            "of features in the dataset will be used without alteration."
+        ),
+    )
+
+    def _as_numpy_dtype(type_str):
+        return np.dtype(type_str).type
+
+    argparser.add_argument(
+        "--dtype",
+        default="float32",
+        choices=[np.float32, np.float64],
+        type=_as_numpy_dtype,
+        help="Floating points precision.",
+    )
+
+    argparser.add_argument(
+        "--random-state", default=123, type=int, help="Seed set for reproducibility."
+    )
+
+    # NB: this option is currently unreliable, estimators seem to usually return close
+    # results except for a few data points that have significant differences.
+    argparser.add_argument(
+        "--run-consistency-checks",
+        action="store_true",
+        help="Check that all the estimators in the benchmark return the same results.",
+    )
+
+    class _IncludeAll:
+        def __contains__(self, item):
+            return True
+
+    argparser.add_argument(
+        "--engine-providers",
+        nargs="+",
+        choices=["scikit-learn", "daal4py", "kmeans-dpcpp", "sklearn-numba-dpex"],
+        default=_IncludeAll(),
+        help=(
+            "Only run the selected engine providers. If not specified, include all "
+            "providers in the benchmark."
+        ),
+    )
+
+    argparser.add_argument(
+        "--include-slow",
+        action="store_true",
+        help="Include in the benchmark the engine providers that are known to be slow.",
+    )
```

Comment:
Why do not we use `skip_slow`?

Reply:
Because we want `skip_slow` to be true by default, and `--include-slow` rings nicer than `--skip-slow False` or `--not-skip-slow`